kong/pdk/node.lua

```diff
@@ -255,6 +257,23 @@ local function new(self)
     return gsub(hostname, "\n$", "")
   end
 
+
+  local prefix = self and self.configuration and self.configuration.prefix
+  if prefix and self.configuration.role == "data_plane" then
+    local id, err = private_node.load_node_id(prefix)
+    if id then
+      node_id = id
+      ngx.log(ngx.DEBUG, "restored node_id from the filesystem: ", node_id)
+
+    else
+      id = _NODE.get_id()
```

Comment:
Should this be `node_id = _NODE.get_id()`?

Reply:
@flrgh The var node_id will be set after call _NODE.get_id().
```lua
function _NODE.get_id()
  -- ...
  node_id, err = shm:get(NODE_ID_KEY)
  -- ...
end
```